### ISSUE TITLE
False positive for `no-changed-when` rule

### ISSUE THREAD

**Issue by yellowhat** [ansible/ansible-lint#3069]
##### Summary

Hi,
the release 6.13.1 includes the PR #3050.
Now I get a violation of the `no-changed-when` rule for the following:

```yaml
    - name: Swap - off
      ansible.builtin.command:
        cmd: swapoff -a
      when: ansible_swaptotal_mb > 0
``` 

From my understading, the `command` should be idempotent by the `when` key. 

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
6.13.1
```

```bash
pip install ansible-lint==6.13.1
```

##### STEPS TO REPRODUCE

```console
ansible-lint *.yaml
```

##### Desired Behavior

Rule followed

##### Actual Behavior

Rule forbidden

**Comment by ssbarnea** [comment]
@yellowhat "when" determines only if a task runs or not and not if it is idempotent or not when it runs. The only way to make ansible know if the task modified something or not when it did run is to specify `changed_when`. Do it, and this rule will not trigger.
